src/components/Nav/index.tsx

```diff
@@ -62,52 +63,48 @@ const Nav: FC<IProps> = ({ path }) => {
           </Link>
           {/* Desktop */}
           <Flex
-            justifyContent="space-between"
-            w="100%"
-            display={{ base: "none", lg: "flex" }}
+            w="full"
+            justifyContent={{ base: "flex-end", lg: "space-between" }}
             ml={{ base: 3, xl: 8 }}
           >
-            <Menu path={path} sections={linkSections} />
+            <Menu hideBelow="lg" path={path} sections={linkSections} />
             <Flex
               alignItems="center"
               justifyContent="space-between"
-              gap={{ base: 1, xl: 0 }}
+              gap={{ base: 2, xl: 4 }}
             >
-              <Search ref={searchRef} />
-              <IconButton
-                aria-label={
-                  isDarkTheme ? "Switch to Light Theme" : "Switch to Dark Theme"
-                }
-                icon={<Icon as={isDarkTheme ? MdWbSunny : MdBrightness2} />}
-                variant="icon"
-                size="sm"
-                fontSize="2xl"
-                ms={{ xl: 2 }}
-                _hover={{ color: "primary.base" }}
-                onClick={toggleColorMode}
+              <Search {...searchModalDisclosure} />
+              {/* Mobile */}
+              <MobileNavMenu
+                {...mobileNavProps}
+                hideFrom="lg"
+                toggleSearch={searchModalDisclosure.onOpen}
+                drawerContainerRef={navWrapperRef}
               />
-              <ButtonLink
-                to={`/languages/${fromPageParameter}`}
-                variant="icon"
-                px={{ base: 1, xl: 1.5 }}
-                size="sm"
-                fontSize="md"
-              >
-                <Icon as={MdLanguage} fontSize="2xl" />
-                <Text as="span" pl={2}>
-                  <Box as="span" hideBelow="lg">
-                    {t("languages")}
-                  </Box>{" "}
-                  {i18n.language.toUpperCase()}
-                </Text>
-              </ButtonLink>
+              <HStack spacing={2} hideBelow="lg">
+                <Button
+                  aria-label={
+                    isDarkTheme
+                      ? "Switch to Light Theme"
+                      : "Switch to Dark Theme"
+                  }
+                  variant="secondaryGhost"
+                  px={1.5}
+                  onClick={toggleColorMode}
+                >
+                  <Icon as={isDarkTheme ? MdWbSunny : MdBrightness2} />
+                </Button>
```

Comment:
Using `IconButton` also works here, and still simplified from the original
```suggestion
                <IconButton
                  icon={isDarkTheme ? <MdWbSunny /> : <MdBrightness2 />}
                  aria-label={
                    isDarkTheme
                      ? "Switch to Light Theme"
                      : "Switch to Dark Theme"
                  }
                  variant="secondaryGhost"
                  px={1.5}
                  onClick={toggleColorMode}
                />
```

Reply:
I'm not sure why it would be doing that. I'm not seeing that issue when I try to apply it. Maybe something to follow up with later once this PR gets pushed?

Reply:
Hmm probably I was just doing silly things and it works fine xD

Yea...lets leave it for now and we can tackle it later.